web/src/main/java/org/cbioportal/web/CoExpressionController.java

```diff
@@ -34,27 +33,40 @@
     private CoExpressionService coExpressionService;
 
     @PreAuthorize("hasPermission(#molecularProfileId, 'MolecularProfile', 'read')")
-    @RequestMapping(value = "/molecular-profiles/{molecularProfileId}/co-expressions/fetch",
+    @RequestMapping(value = "/molecular-profiles/{molecularProfileIdA}/{molecularProfileIdB}/co-expressions/fetch",
         method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
         produces = MediaType.APPLICATION_JSON_VALUE)
-    @ApiOperation("Fetch co-expressions in a molecular profile")
+    @ApiOperation("Calculates correlations between a genetic entity from a specific profile and another profile from the same study")
     public ResponseEntity<List<CoExpression>> fetchCoExpressions(
-        @ApiParam(required = true, value = "Molecular Profile ID e.g. acc_tcga_rna_seq_v2_mrna")
-        @PathVariable String molecularProfileId,
-        @ApiParam(required = true, value = "List of Sample IDs/Sample List ID")
+        @ApiParam(required = true, value = "Molecular Profile ID from the Genetic Entity referenced in the co-expression filter e.g. acc_tcga_rna_seq_v2_mrna")
+        @PathVariable String molecularProfileIdA,
```

Comment:
As I said before, these 2 should be @RequestParam because multiple resources back to back in the path would indicate that they have a parent-child relationship, but it's not the case here. So the path should be `/molecular-profiles/co-expressions/fetch`